xwiki-commons-core/xwiki-commons-job/xwiki-commons-job-default/src/main/java/org/xwiki/job/internal/Version3JobStatusFolderResolver.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.job.internal;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.annotation.Priority;
+import javax.inject.Named;
+
+import jakarta.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+
+/**
+ * Version 3 of {@link JobStatusFolderResolver} that uses URL encoding but with length limit and additional encoding
+ * for reserved characters. Introduced in XWiki 16.10.6 and 17.2.0.
+ *
+ * @version $Id$
+ * @since 17.2.0RC1
+ * @since 16.10.6
+ */
+@Component
+@Singleton
+@Named("version3")
```

Comment:
I'm not sure we really need to put a prefix here. The number as a String should be fine.

Reply:
We need some prefix in the class name and to me, it felt more consistent to have the same prefix in the hint.